File: JUNGOL/Beginner/#5945 숫자 삼각형1/숫자 삼각형1.py
def main():
    n = int(input())

    if n<1 or n>50 or n%2==0: print("INPUT ERROR!")
    else:
        num,dir = 1,1
        for i in range(1, n+1):
            if dir == 1:
                for j in range(i):
                    print(num, end=' ')
                    num += 1
                print()
            else:
                for j in range(num+i-1, num-1, -1):
                    print(j, end=' ')
                num += i
                print()

            dir = -dir

main()

File: JUNGOL/Beginner/#5945 숫자 삼각형1/숫자 삼각형1.cpp
#include <iostream>
using namespace std;

int main() {
	int n; cin>>n;

	if(n<1 || n>50 || !(n%2)) cout<<"INPUT ERROR!\n";
	else{
		int num=1, dir=1;
		for(int i=1; i<=n; i++){
			if(dir == 1){
				for(int j=0; j<i; j++) printf("%d ", num++);
				cout<<endl;
			}
			else{
				for(int j=num+i-1; j>=num; printf("%d ", j--));
				num += i;
				cout<<endl;
			}

            dir = -dir;
		}
	}
}
